server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java

```diff
@@ -307,11 +332,49 @@ private CjpqKey addJobToQueue(TabletMetadata tabletMetadata, CompactionJob job)
 
     var key = new CjpqKey(job);
     jobQueue.put(key, new CompactionJobQueues.MetaJob(job, tabletMetadata));
+    jobAges.computeIfAbsent(tabletMetadata.getExtent(), e -> Timer.startNew());
```

Comment:
Instead of adding here

```suggestion
```

Could add at the following location.  This will only be done once for the tablet.  Also can handle the edge case of the queue being full at that location in the code.

```diff
diff --git a/server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java b/server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java
index e4e1f4ca02..b55172276e 100644
--- a/server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java
+++ b/server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java
@@ -213,6 +213,9 @@ public class CompactionJobPriorityQueue {
     if (!newEntries.isEmpty()) {
       checkState(tabletJobs.put(tabletMetadata.getExtent(), new TabletJobs(generation, newEntries))
           == null);
+      jobAges.computeIfAbsent(tabletMetadata.getExtent(), e->Timer.startNew());
+    }else{
+      jobAges.remove(tabletMetadata.getExtent());
     }
```